src/components/UserPost/UserPost.jsx

```diff
@@ -1,11 +1,91 @@
+import { useState } from "react"
+import {
+    AiFillLike
+} from "../../assets/react-icons"
 import './UserPost.css'
 
-function UserPost()
+function UserPost({userPostDetails})
 {
-    //This is temporary layout, will add user post content to it in future PRs
+    
+    const {
+        contentText,
+        userName,
+        userProfilePic,
+        noOfLikes
+    } = userPostDetails;
+
+    const [ showFullText, setShowFullText ] = useState(false)
+    let postText1 = "", postText2 = "";
+    
+    let postTextArray = contentText.split(" ")
+
+    if(postTextArray.length<45)
+    {
+        postText1 = contentText;
+    }
+    else
+    {
+        postText1 = postTextArray.slice(0,46).join(" ")
+        postText2 = postTextArray.slice(46).join(" ")
+    }
+
+
     return (
         <div className='user-post'>
-            
+            <div className="user-post-header">
+                <div className="post-header-left-container">
+                    <div className="avatar avatar-x-small">
+                        <img 
+                            className="avatar-img" 
+                            src={userProfilePic===""
```

Comment:
Suggestion: i think a better way you could do this was giving a default value to userProfilePic on line 13 itself if some value is received it will replace the default one

Reply:
Yeah, you are right, made changes, thanks for suggestion :D